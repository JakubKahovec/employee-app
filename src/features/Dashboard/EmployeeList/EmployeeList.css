.Employee-list-table-striped-rows tr:nth-child(2n) td {
  background-color: #fbfbfb;
}
.Employee-list-table-striped-rows thead {
  background-color: #f1f1f1;
}
