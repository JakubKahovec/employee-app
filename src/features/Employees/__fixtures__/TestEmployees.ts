import { Employee } from "AppModel";

export const employee1: Employee = {
  jobTitle: "CEO",
  gender: "Male",
  tenure: "8",
  name: "John Green",
};

export const employee2: Employee = {
  jobTitle: "CTO",
  gender: "Female",
  tenure: "7",
  name: "Sheila Pink",
};

export const employee3: Employee = {
  jobTitle: "CFO",
  gender: "Female",
  tenure: "6",
  name: "Ema Shelton",
};

export const employee4: Employee = {
  jobTitle: "CFO",
  gender: "Male",
  tenure: "3",
  name: "Peter Holmes",
};
