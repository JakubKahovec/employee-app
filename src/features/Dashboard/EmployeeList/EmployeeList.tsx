import { Employee } from "AppModel";
import { Table } from "antd";
import { ColumnProps } from "antd/es/table";
import * as React from "react";
import "./EmployeeList.css";

interface EmployeeListProps {
  employees: Employee[];
  isLoading: boolean;
}

export class EmployeeList extends React.Component<EmployeeListProps> {
  render() {
    const { isLoading, employees } = this.props;
    return (
      <Table<Employee>
        pagination={false}
        scroll={{ y: 400 }}
        className="Employee-list-table-striped-rows"
        loading={isLoading}
        rowKey={(_, index) => `${index}`}
        columns={this.columns()}
        size="small"
        dataSource={employees}
        footer={() => <span>Total employees: {employees.length}</span>}
      />
    );
  }

  private columns(): Array<ColumnProps<Employee>> {
    return [
      {
        title: "Name",
        width: 200,
        key: "name",
        dataIndex: "name",
        sorter: (e1, e2) => e1.name.localeCompare(e2.name),
      },
      {
        title: "Job Title",
        width: 200,
        key: "jobTitle",
        dataIndex: "jobTitle",
        sorter: (e1, e2) => e1.jobTitle.localeCompare(e2.jobTitle),
      },
      {
        title: "Tenure",
        width: 50,
        key: "tenure",
        dataIndex: "tenure",
        sorter: (e1, e2) => Number(e1.tenure) - Number(e2.tenure),
      },
      {
        title: "Gender",
        width: 200,
        key: "gender",
        dataIndex: "gender",
        sorter: (e1, e2) => e1.gender.localeCompare(e2.gender),
        render: (gender: string) => gender.substr(0, 1),
      },
    ];
  }
}
