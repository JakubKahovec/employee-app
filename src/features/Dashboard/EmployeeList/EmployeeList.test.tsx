import * as React from "react";

import { createRenderer } from "react-test-renderer/shallow";
import { employee1, employee2, employee3 } from "../../Employees/__fixtures__/TestEmployees";
import { EmployeeList } from "./EmployeeList";

it("renders correctly EmployeeList", () => {
  const renderer = createRenderer();
  const tree = renderer.render(<EmployeeList employees={[employee1, employee2, employee3]} isLoading={true} />);
  expect(tree).toMatchSnapshot();
});
