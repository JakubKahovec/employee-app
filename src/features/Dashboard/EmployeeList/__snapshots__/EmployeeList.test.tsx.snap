// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly EmployeeList 1`] = `
<Table
  className="Employee-list-table-striped-rows"
  columns={
    Array [
      Object {
        "dataIndex": "name",
        "key": "name",
        "sorter": [Function],
        "title": "Name",
        "width": 200,
      },
      Object {
        "dataIndex": "jobTitle",
        "key": "jobTitle",
        "sorter": [Function],
        "title": "Job Title",
        "width": 200,
      },
      Object {
        "dataIndex": "tenure",
        "key": "tenure",
        "sorter": [Function],
        "title": "Tenure",
        "width": 50,
      },
      Object {
        "dataIndex": "gender",
        "key": "gender",
        "render": [Function],
        "sorter": [Function],
        "title": "Gender",
        "width": 200,
      },
    ]
  }
  dataSource={
    Array [
      Object {
        "gender": "Male",
        "jobTitle": "CEO",
        "name": "John Green",
        "tenure": "8",
      },
      Object {
        "gender": "Female",
        "jobTitle": "CTO",
        "name": "Sheila Pink",
        "tenure": "7",
      },
      Object {
        "gender": "Female",
        "jobTitle": "CFO",
        "name": "Ema Shelton",
        "tenure": "6",
      },
    ]
  }
  footer={[Function]}
  loading={true}
  pagination={false}
  rowKey={[Function]}
  scroll={
    Object {
      "y": 400,
    }
  }
  size="small"
/>
`;
